model/models.py

```diff
@@ -38,16 +40,38 @@ def __init__(self, edge_index, edge_type, num_rel, params=None):
 
 		self.register_parameter('bias', Parameter(torch.zeros(self.p.num_ent)))
 
-	def forward_base(self, sub, rel, drop1, drop2):
-
-		r	= self.init_rel if self.p.score_func != 'transe' else torch.cat([self.init_rel, -self.init_rel], dim=0)
-		x, r	= self.conv1(self.init_embed, self.edge_index, self.edge_type, rel_embed=r)
-		x	= drop1(x)
-		x, r	= self.conv2(x, self.edge_index, self.edge_type, rel_embed=r) 	if self.p.gcn_layer == 2 else (x, r)
-		x	= drop2(x) 							if self.p.gcn_layer == 2 else x
+	def forward_base(self, sub, rel, drop1, drop2, disable_gnn_encoder: bool):
+		"""
+
+		Parameters
+		----------
+		sub subset
+		rel relation
+		drop1 first drop function
+		drop2 second drop function
+		disable_gnn_encoder how to disable the gnn encoder
+
+		Returns subset of the embedding, subset of the relation, embedding
+		-------
+
+		"""
```

Comment:
I like the docu, but do we want to do it on top of the function or in the function itself?

Reply:
As far as I'm aware docstrings in python are always in the function, see here for the official convention: https://peps.python.org/pep-0257/